### PR TITLE
Adds a listener to Enketo to notify on change

### PR DESCRIPTION
**Pull request by garethbowen**
Enketo has an event that fires when the form content has been updated
by the user. This change leverages that so that the "may lost content"
warning dialog is only shown when the user has actually changed a
value.

This fixes a bug where the warning is shown when the form failed to
render, and also a bug where the warning is shown when the form has no
fields.

medic/medic-webapp#1601
medic/medic-webapp#3045

### CODE REVIEW

File: static/js/controllers/reports-add.js [hunk 1]
```diff
@@ -70,7 +74,8 @@ angular.module('inboxControllers').controller('ReportsAddCtrl',
           getReportContent(model.doc),
           XmlForm(model.formInternalId, { include_docs: true })
         ]).then(function(results) {
-          Enketo.render('#report-form', results[1].id, results[0])
+          $scope.enketoStatus.edited = false;
```
Comment (alxndrsn) [inline static/js/controllers/reports-add.js]: Why is this required here, but not in the other controllers?

Comment (garethbowen) [inline static/js/controllers/reports-add.js]: Ouch. It's needed - I missed it. Thanks.
